lucene/misc/src/java/org/apache/lucene/misc/store/DirectIODirectory.java

```diff
@@ -154,14 +146,43 @@ public IndexOutput createOutput(String name, IOContext context) throws IOExcepti
     if (context.context != Context.MERGE || context.mergeInfo.estimatedMergeBytes < minBytesDirect) {
       return delegate.createOutput(name, context);
     } else {
-      return new NativeUnixIndexOutput(getDirectory().resolve(name), name, mergeBufferSize);
+      return new DirectIOIndexOutput(getDirectory().resolve(name), name, mergeBufferSize);
     }
   }
 
-  @SuppressForbidden(reason = "java.io.File: native API requires old-style FileDescriptor")
-  private final static class NativeUnixIndexOutput extends IndexOutput {
+  @Override
+  public void deleteFile(String name) throws IOException {
+    delegate.deleteFile(name);
+  }
+
+  @Override
+  public Set<String> getPendingDeletions() throws IOException {
```

Comment:
I wonder if we should simply extend `FilterDirectory` here?  Let that class (designed for such delegation) handle delegating all existing `Directory` methods?  It would lower the risk that a new `Directory` method fails to get delegated here ...

But, we can do this as followon issue.

Reply:
Ha I had this same thought for a split second before, but then got distracted and forgot to come back to it later.  I just pushed a commit to make this change.

Reply:
Great, thanks!